### PR TITLE
Connect API to the Bookstore APP

### PR DESCRIPTION
**Pull request by Kaweesi-Matia**
## Below is the description of what is included in the pull request

- [ ] Refactor code to use Bookstore API.
- [ ] Use Axios with thunk for making API calls.
- [ ] Refactor addBook and removeBook to return functions.
- [ ] Dispatch actions that update the state

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,34 +1,48 @@
...
+  try {
+    await axios.delete(`${process.env.REACT_APP_BASE_URL}/${id}`);
+    return dispatch({ type: REMOVE_BOOK, id });
+  } catch (err) { return err; }
+};
+
+export const getBooksFunction = () => async (dispatch) => {
+  try {
+    const response = await axios.get(`${process.env.REACT_APP_BASE_URL}`);
+    return dispatch({ type: GET_BOOKS, payload: response.data });
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=Use%20fetch%20or%20axios%20with%20createAsyncThunk%20for%20making%20API%20calls.), Please make sure that `createAsyncThunk` is used in the project for creating thunk action creators.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Kaweesi-Matia,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Kaweesi-Matia** [comment]
Add createAsyncThunk

**Review by m05e5 — APPROVED** [review]
Hi @Kaweesi-Matia , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**